### PR TITLE
chore: add dev debug build using delve

### PR DESCRIPTION
**Pull request by bradleyjones**
Skaffold debug doesn't currently support go 1.21 so this is a workaround
until skaffold is updated.

Signed-off-by: Bradley Jones <bradley.jones@anchore.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -80,6 +80,26 @@ lint-fix: ## Auto-format all source code + run golangci lint fixers
 build:
 	$(TEMPDIR)/goreleaser build --clean --snapshot
 
+.PHONE: debug
```
Comment (asomya) [inline Makefile:83]: typo .PHONE instead of .PHONY

**Review by asomya — CHANGES_REQUESTED** [review]
minor nit, LGTM otherwise

File: Makefile [hunk 2]
```diff
@@ -80,6 +80,26 @@ lint-fix: ## Auto-format all source code + run golangci lint fixers
...
+debug: debug-build debug-run
+
+.PHONY: debug-build
+debug-build:
+	docker build -f Dockerfile-dev -t anchore/harbor-scanner-adapter:debug .
+
+.PHONY: debug-stop
+debug-stop:
+	kubectl delete $(shell kubectl get pods -o name | grep harbor-scanner-anchore)
```
Comment (asomya) [inline Makefile:92]: is it possible to use a label selector instead of relying on grep?

Comment (bradleyjones) [reply Makefile:92]: yeah good idea

File: Makefile [hunk 3]
```diff
@@ -80,6 +80,26 @@ lint-fix: ## Auto-format all source code + run golangci lint fixers
...
+	docker build -f Dockerfile-dev -t anchore/harbor-scanner-adapter:debug .
+
+.PHONY: debug-stop
+debug-stop:
+	kubectl delete -l app=harbor-scanner-anchore
+
+.PHONY: debug-run
+debug-run:
+	kubectl apply -f ./k8s/harbor-adapter-anchore-debug.yaml
+	kubectl port-forward $(shell kubectl get pods -o name | grep harbor-scanner-anchore) 2345:2345 8080:8080
```
Comment (asomya) [inline Makefile:97]: kubectl get pods  -l app=harbor-scanner-anchore

File: Dockerfile-dev [hunk 4]
```diff
@@ -0,0 +1,17 @@
+########## Builder ##########
+FROM golang:1.21-alpine AS builder
+
+# Install the latest version of Delve
+RUN go install github.com/go-delve/delve/cmd/dlv@latest
```
Comment (juanjsebgarcia) [inline Dockerfile-dev:5]: This seems like it could break if they change something upstream, sure you don't want to pin it?

**Review by juanjsebgarcia — APPROVED** [review]
1 comment but non-blocking as it maybe what you intended

File: Dockerfile-dev (hunk 4 above)
Comment (bradleyjones) [reply Dockerfile-dev:5]: they're supposed to be backwards compatible but yeah that's a good idea can't trust it will be. I'll update quickly

**Pull request merged**